### ISSUE TITLE
Error compiling translated sources that have a static field named "initialize".

### ISSUE THREAD

**Issue by AndreaPQ** [google/j2objc#1664]
We ran into this problem with a third party library, the sample code below reproduces the case :

```Java
final class State 
{
    public static final int initialize=0;	
    public static final int ready=1;
    public static final int anotherState=2;
}
```
```
j2objc -l  Issue.java 
translating Issue.java
Translated 1 file: 0 errors, 0 warnings

j2objcc -c issue.m
issue.m:54:1: error: called object type 'int' is not a function or function pointer
J2OBJC_CLASS_TYPE_LITERAL_SOURCE(State)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/j2objc/j2objc-2.8/include/J2ObjC_common.h:207:22: note: expanded from macro
      'J2OBJC_CLASS_TYPE_LITERAL_SOURCE'
    TYPE##_initialize(); \
    ~~~~~~~~~~~~~~~~~^
1 error generated.
```

The problem seems caused by the double definition of the symbol "State_initialize"  on the generated header file **Issue.h**, one definition is produced by the macro **J2OBJC_EMPTY_STATIC_INIT**(State) and the other one for the constant value "public static final int initialize=0;"  

```objective-c
J2OBJC_EMPTY_STATIC_INIT(State)	// -> __attribute__((always_inline)) inline void State_initialize() {}
..
#define State_initialize 0
```

In the **Issue.m** generated file, the macro J2OBJC_CLASS_TYPE_LITERAL_SOURCE(State) cause the compilation error in the point where it handle the "State_initialize" symbol as a function,  

```objective-c
// J2OBJC_CLASS_TYPE_LITERAL_SOURCE(State) becomes
IOSClass *State_class_() { 
	static IOSClass *cls; 
	static dispatch_once_t token; 
	State_initialize(); 			// <--- Error
	dispatch_once(&token, ^{ cls = IOSClass_fromClass([State class]); }); 
	return cls; 
}
```

**Comment by tomball** [comment]
The problem is that static fields have accessor functions named "<type-name>_<field-name>" and all classes have an initialization function named "<type-name>_initialize", so the accessor for an "initialize" constant will clash with its class initialize function. I'm surprised no one has run into this before!

Fix is out for a code review. The workaround is to name "initialize" to anything else, like "Initialize", "INITIALIZE", "initial", etc.

**Comment by tomball** [comment]
Fixed in [41ecaf4](https://github.com/google/j2objc/commit/41ecaf407292bb80fb3e64c10ef392169ebbb706). Thanks for the bug report.
